core/opendaq/reader/src/block_reader_impl.cpp

```diff
@@ -9,48 +9,72 @@ BEGIN_NAMESPACE_OPENDAQ
-                                 SampleType valueReadType,
-                                 SampleType domainReadType,
-                                 ReadMode mode)
+static SizeT calculateWrittenSamplesCount(SizeT initialWrittenSamplesCount, SizeT finalWrittenSamplesCount)
+{
+    auto writtenSamplesCount = finalWrittenSamplesCount > initialWrittenSamplesCount
+                                   ? finalWrittenSamplesCount - initialWrittenSamplesCount
+                                   : initialWrittenSamplesCount - finalWrittenSamplesCount;  // in case of counter wrapped
+    return writtenSamplesCount;
+};
+
+BlockReaderImpl::BlockReaderImpl(
+    const SignalPtr& signal, SizeT blockSize, SizeT overlap, SampleType valueReadType, SampleType domainReadType, ReadMode mode)
     : Super(signal, mode, valueReadType, domainReadType)
     , blockSize(blockSize)
+    , overlap(overlap)
 {
+    if (this->overlap >= 100)
+        throw InvalidParameterException("Overlap could not be greater or equal 100%");
+
+    overlappedBlockSize = (blockSize * overlap) / 100;
+    overlappedBlockSizeRemainder = blockSize - overlappedBlockSize;
+
     port.setNotificationMethod(PacketReadyNotification::SameThread);
     readDescriptorFromPort();
 }
 
-BlockReaderImpl::BlockReaderImpl(IInputPortConfig* port,
-                                 SizeT blockSize,
-                                 SampleType valueReadType,
-                                 SampleType domainReadType,
-                                 ReadMode mode)
+BlockReaderImpl::BlockReaderImpl(
+    IInputPortConfig* port, SizeT blockSize, SizeT overlap, SampleType valueReadType, SampleType domainReadType, ReadMode mode)
     : Super(InputPortConfigPtr(port), mode, valueReadType, domainReadType)
     , blockSize(blockSize)
+    , overlap(overlap)
 {
+    if (this->overlap >= 100)
+        throw InvalidParameterException("Overlap could not be greater or equal 100%");
+
+    overlappedBlockSize = (blockSize * overlap) / 100;
+    overlappedBlockSizeRemainder = blockSize - overlappedBlockSize;
+
     this->port.setNotificationMethod(PacketReadyNotification::Scheduler);
 }
 
-BlockReaderImpl::BlockReaderImpl(const ReaderConfigPtr& readerConfig,
-                                 SampleType valueReadType,
-                                 SampleType domainReadType,
-                                 SizeT blockSize,
-                                 ReadMode mode)
+BlockReaderImpl::BlockReaderImpl(
+    const ReaderConfigPtr& readerConfig, SampleType valueReadType, SampleType domainReadType, SizeT blockSize, SizeT overlap, ReadMode mode)
     : Super(readerConfig, mode, valueReadType, domainReadType)
     , blockSize(blockSize)
+    , overlap(overlap)
 {
+    if (this->overlap >= 100)
+        throw InvalidParameterException("Overlap could not be greater or equal 100%");
+
+    overlappedBlockSize = (blockSize * overlap) / 100;
+    overlappedBlockSizeRemainder = blockSize - overlappedBlockSize;
```

Comment:
This could be refactored into a separate function, as it's often repeated.

Reply:
Done